src/core/dep-graph.ts

```diff
@@ -138,6 +138,11 @@ class DepGraphImpl implements types.DepGraphInternal {
     return count;
   }
 
+  public equals(other: types.DepGraph, compareRoot: boolean = false): boolean {
+    const otherDepGraph = other as types.DepGraphInternal;
```

Comment:
following @gjvis comments here: https://github.com/snyk/dep-graph/pull/26#discussion_r295833037, suggest instead doing what you started from :) 
```typescript
const otherDepGraph = fromJSON(other.toJSON());
```